pkg/operator/ceph/cluster/osd/osd.go

```diff
@@ -286,9 +291,22 @@ func deploymentOnPVC(c *Cluster, osd OSDInfo, pvcName string, config *provisionC
 		return nil, errors.Wrapf(err, "failed to generate deployment for %s", osdLongName)
 	}
 
+	err = prepareForDeployment(c, osdProps, osd, osdLongName)
+	if err != nil {
+		return nil, errors.Wrapf(err, "failed to prepare for deployment %s", osdLongName)
+	}
+
 	return d, nil
 }
 
+func prepareForDeployment(c *Cluster, osdProps osdProperties, osd OSDInfo, osdLongName string) error {
```

Comment:
It feels unnecessary to pass `osdLongName`since you have `osd OSDInfo`, you can build it directly within `prepareForDeployment()`